jib-core/src/main/java/com/google/cloud/tools/jib/cache/CacheStorageWriter.java

```diff
@@ -81,15 +82,27 @@ private static void moveIfDoesNotExist(Path source, Path destination) throws IOE
       return;
     }
 
+    // Some Windows users report java.nio.file.AccessDeniedException that we suspect is caused
+    // by anti-virus programs, like Windows Defender, that open new files for scanning.
+    // Retry the rename up to 5 times.
+    IOException exception = null;
     try {
-      Files.move(source, destination);
-
-    } catch (FileSystemException ex) {
-      if (!Files.exists(destination)) {
-        // TODO to log that the destination exists
-        throw ex;
+      boolean success =
+          Retry.action(() -> Files.move(source, destination))
+              .until(() -> Files.exists(destination))
+              .maximumRetries(5)
+              .retryOnException(ex -> ex instanceof FileSystemException)
+              .sleep(15, TimeUnit.MILLISECONDS)
+              .run();
+      if (success) {
+        return;
       }
+    } catch (IOException ex) {
+      exception = ex;
```

Comment:
I think you can throw the exception directly here. There's no looping. Calling `run()` will either throw an exception or not.

Reply:
+1